### PR TITLE
Add ir builder

### PR DESCRIPTION
**Pull request by Trigger2000**
/cc @SlynkoDenis

### CODE REVIEW

File: ir/inst.h [hunk 1]
```diff
@@ -3,23 +3,70 @@
...
+
+    void Dump();
+
+private:
+    Inst(Opcode opcode, std::vector<VReg*> inputs) : opcode_(opcode), inputs_(inputs) {
+        ++max_inst_id_;
+        inst_id_ = max_inst_id_;
+    }
+
+    void DumpOpcode()
```
Comment (SlynkoDenis) [inline ir/inst.h:48]: Can be implemented by creating a `constexpr std::array<const char *, N> names{...}` and getting the name with `names[static_cast<int>(opcode)]`

File: ir/function.h [hunk 2]
```diff
@@ -1,12 +1,102 @@
...
 class Function {
 public:
-    std::vector<Inst> insts;
-    std::vector<VRegType> vregs;
+    Function(std::string name) : name_(name) {}
+    void AddInst(Inst* inst)
+    {
+        inst->SetFunction(this);
+        for (auto input: inst->GetInputs()) {
+            assert(input != nullptr);
```
Comment (SlynkoDenis) [inline ir/function.h:20]: Consider replacing `assert` with custom macros `ASSERT` (see `macros.h`) so it could be removed in `RELEASE` build type

Comment (Trigger2000) [reply ir/function.h:20]: Actually, standard assert from <cassert> is also removed in release build. But for consistency can replace it with ASSERT

File: ir/inst.h (hunk 1 above)
Comment (Trigger2000) [reply ir/inst.h:48]: Yes, sure. But I suggest to do it in further versions

**Pull request merged**